### PR TITLE
NONPR-1717-Play 2.7-upgraded dependencies and added it:tests

### PR DESCRIPTION
**Pull request by Leeroyc720**
**Changes**

- upgraded bootstrap version to the latest play 27
- added it: tests to support the upgrade changes with http-verbs (contractSpec & XApiHeaderIntegrationSpec)
- added X-API-Key it: tests to assert tests with http-verbs included and for assurance that we haven't changed the behavior
- updated stride-integration file
- tested E2E with the backend service 

**Acceptance -tests results:** 

- 24 Scenarios (24 passed)
- 196 Steps (196 passed)
- 3m51.712s
- [info] Passed: Total 186, Failed 0, Errors 0, Passed 186
- [success] Total time: 249 s, completed 8 Jun 2021, 10:28:32

**Integration-Tests results**

- [info] Tests: succeeded 21, failed 0, canceled 0, ignored 0, pending 0
- [info] All tests passed.
- [info] Passed: Total 21, Failed 0, Errors 0, Passed 21
- [success] Total time: 20 s, completed 8 Jun 2021, 10:32:22

**Tests results**
- [info] Tests: succeeded 52, failed 0, canceled 0, ignored 0, pending 0
- [info] All tests passed.
- [info] Passed: Total 52, Failed 0, Errors 0, Passed 52
- [success] Total time: 13 s, completed 8 Jun 2021, 10:38:36

### CODE REVIEW

File: app/Module.scala [hunk 1]
```diff
@@ -41,8 +41,6 @@ class Module(val environment: Environment, val configuration: Configuration) ext
     bind(classOf[HttpGet]).to(classOf[HttpVerbs])
     bind(classOf[HttpPost]).to(classOf[HttpVerbs])
     bind(classOf[NrsRetrievalConnector]).to(classOf[NrsRetrievalConnectorImpl])
-    bind(classOf[AuthConnector]).to(classOf[MicroAuthConnector])
```
Comment (PaulHodgson) [inline app/Module.scala:44]: the play upgrade broke stride auth integration; after fixing this it seemed that `MicroAuthConnector` was no longer needed as an `AuthConnector` is already provided

File: app/config/HttpHead.scala [hunk 2]
```diff
@@ -27,39 +27,37 @@ import uk.gov.hmrc.play.http.ws.{WSHttpResponse, WSRequest}
...
 
 trait WSHead extends WSRequest with CoreHead with HeadHttpTransport {
 
-  override def doHead(url: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
-    import scala.concurrent.ExecutionContext.Implicits.global
-
-    buildRequest(url).head().map(new WSHttpResponse(_))
+  override def doHead(url: String, headers: Seq[(String, String)])
+                     (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
+    buildRequest(url, headers).head().map(WSHttpResponse(_))
```
Comment (PaulHodgson) [inline app/config/HttpHead.scala:47]: explicit headers are now required, which drives a number of changes.

File: conf/application.conf [hunk 3]
```diff
@@ -34,6 +34,8 @@ play.modules.enabled += "uk.gov.hmrc.play.bootstrap.graphite.GraphiteMetricsModu
 
 # Provides an implementation and configures all filters required by a Platform frontend microservice.
 play.modules.enabled += "uk.gov.hmrc.play.bootstrap.frontend.FrontendModule"
+play.modules.enabled += "uk.gov.hmrc.play.bootstrap.HttpClientModule"
```
Comment (PaulHodgson) [inline conf/application.conf:37]: both modules added to fix the stride auth integration. HttpClientModule seems required for AuthModule

File: it/uk/gov/hmrc/nrsretrievalfrontend/XApiHeaderIntegrationSpec.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nrsretrievalfrontend
+
+import play.api.libs.ws.WSClient
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.nrsretrievalfrontend.stubs.NrsRetrievalStubs._
+
+class XApiHeaderIntegrationSpec extends IntegrationSpec {
```
Comment (PaulHodgson) [inline it/uk/gov/hmrc/nrsretrievalfrontend/XApiHeaderIntegrationSpec.scala:23]: Header logic is split between the controller and connector so is tested in an integration spec.

File: it/uk/gov/hmrc/nrsretrievalfrontend/connectors/NrsRetrievalContractSpec.scala [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+import org.joda.time.LocalDate
+import org.scalatest.Assertion
+import play.api.libs.ws.WSResponse
+import uk.gov.hmrc.http.{HeaderCarrier, HttpException, HttpResponse, UpstreamErrorResponse}
+import uk.gov.hmrc.nrsretrievalfrontend.IntegrationSpec
+import uk.gov.hmrc.nrsretrievalfrontend.stubs.NrsRetrievalStubs._
+
+import java.time.ZonedDateTime
+
+class NrsRetrievalContractSpec extends IntegrationSpec {
```
Comment (PaulHodgson) [inline it/uk/gov/hmrc/nrsretrievalfrontend/connectors/NrsRetrievalContractSpec.scala:30]: provides a contract-style spec for interactions with the nrs-retrieval back end service.

perhaps in time we could expand this approach using something like `pact`.

File: test/connectors/NrsRetrievalConnectorSpec.scala [hunk 6]
```diff
@@ -36,34 +35,69 @@ import uk.gov.hmrc.play.audit.http.connector.AuditConnector
 import uk.gov.hmrc.play.audit.model.Audit
 import uk.gov.hmrc.play.test.UnitSpec
 
+import javax.inject.Provider
 import scala.concurrent.Future
 
 class NrsRetrievalConnectorSpec extends UnitSpec with MockitoSugar with NrsSearchFixture with Infrastructure with BeforeAndAfterEach {
+  private val mockWsHttp = mock[WSHttpT]
```
Comment (PaulHodgson) [inline test/connectors/NrsRetrievalConnectorSpec.scala:42]: moved to the beginning of the class to move some IntelliJ warnings.

**Review by greg-gajda — APPROVED** [review]
LGTM

**Pull request merged**